File: benchmarks/cpp/closures.cpp
#include <iostream>
#include <functional>
std::function<long long()> make_counter() {
    long long n = 0;
    return [n]() mutable -> long long { return ++n; };
}
int main() {
    long long total = 0;
    for (int i = 0; i < 1000; i++) {
        auto c = make_counter();
        for (int j = 0; j < 100; j++) total += c();
    }
    std::cout << total << "\n";
}

File: benchmarks/closures.py
def make_counter():
    n = 0
    def counter():
        nonlocal n
        n += 1
        return n
    return counter

total = 0
for i in range(1000):
    c = make_counter()
    for j in range(100):
        total += c()
print(total)
